tests/basic/storage_pool_test.go

```diff
@@ -2053,3 +2209,1705 @@ var _ = Describe("{VolUpdateAddDrive}", func() {
+
+		stepLog := "Initiate add cloud drive and reboot node"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			err = Inst().V.AddCloudDrive(&stNode, deviceSpec, -1)
+			log.FailOnError(err, fmt.Sprintf("Add cloud drive failed on node %s", stNode.Name))
+			time.Sleep(3 * time.Second)
+			err = RebootNodeAndWait(stNode)
+			log.FailOnError(err, fmt.Sprintf("error rebooting node %s", stNode.Name))
+			log.InfoD("Validate pool rebalance after drive add")
+			err = ValidatePoolRebalance()
+			log.FailOnError(err, "Pool re-balance failed")
+			dash.VerifyFatal(err == nil, true, "PX is up after add drive")
+
+			var newTotalPoolSize uint64
+			pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+			log.FailOnError(err, "error getting pools list")
+			dash.VerifyFatal(len(pools) > 0, true, "Verify pools exist")
+			for _, pool := range pools {
+				newTotalPoolSize += pool.GetTotalSize() / units.GiB
+			}
+			dash.VerifyFatal(newTotalPoolSize, expectedTotalPoolSize, fmt.Sprintf("Validate total pool size after add cloud drive on node %s", stNode.Name))
+		})
+		opts := make(map[string]bool)
+		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+		ValidateAndDestroy(contexts, opts)
+
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts, testrailID, runID)
+	})
+
+})
+
+var _ = Describe("{MulPoolsResize}", func() {
+	//1) Deploy px with cloud drive.
+	//2) Select multiple pools
+	//3) Expand multiple pools by resize-disk same time.
+
+	var testrailID = 51291
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/51291
+	var runID int
+	JustBeforeEach(func() {
+		StartTorpedoTest("MulPoolsResize", "Initiate multiple pool resize on same node in parallel ", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+
+	stepLog := "should get the existing storage node with multiple pools and expand pools at same time using resize-disk"
+
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("mulpoolsresiz-%d", i))...)
+		}
+		ValidateApplications(contexts)
+
+		var poolsToBeResized []*api.StoragePool
+
+		pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+		log.FailOnError(err, "Failed to list storage pools")
+		numPoolsToResize := len(pools) / 3
+		i := 1
+		for _, v := range pools {
+			if i == numPoolsToResize {
+				break
+			}
+			poolsToBeResized = append(poolsToBeResized, v)
+			i += 1
+		}
+
+		stepLog = fmt.Sprintf("Expanding multiple pools  on node and pool using resize-disk")
+		Step(stepLog, func() {
+
+			resizedPoolsMap := make(map[string]uint64)
+			for _, selPool := range poolsToBeResized {
+				poolToBeResized, err := GetStoragePoolByUUID(selPool.Uuid)
+				log.FailOnError(err, "Failed to get pool using UUID ")
```

Comment:
Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)